extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -1236,6 +1250,59 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
       (returnedSizeDifferent && heightNotIncreased)
     ) {
       this.attemptChangeSize(newHeight, newWidth).catch(() => {});
+      if (newWidth > width) {
+        this.adjustSlotPostExpansion_(newWidth);
+      }
+    }
+  }
+
+  /**
+   * Ensures that slot is properly centered after being expanded.
+   * @param {number} newWidth The new width of the slot.
+   * @private
+   */
+  adjustSlotPostExpansion_(newWidth) {
+    if (this.flexibleAdSlotData_ == null) {
+      dev().warn(
+        TAG,
+        'Attempted to expand slot without flexible ad slot data.'
+      );
+      return;
+    }
+    const isRtl = isRTL(this.win.document);
+    const dirStr = isRtl ? 'Right' : 'Left';
+    // Guaranteed to be set after exiting if/else.
+    let /** ?number */ newMargin = null;
+    if (newWidth <= this.flexibleAdSlotData_.parentWidth) {
+      // Must center creative within its parent container
+      const {parentStyle} = this.flexibleAdSlotData_;
+      const parentPadding = parseInt(parentStyle[`padding${dirStr}`], 10) || 0;
+      const parentBorder =
+        parseInt(parentStyle[`border${dirStr}Width`], 10) || 0;
+      const whitespace = (this.flexibleAdSlotData_.parentWidth - newWidth) / 2;
+      newMargin = whitespace - parentPadding - parentBorder;
+    } else {
+      // Must center creative within the viewport
+      const viewportWidth = this.getViewport().getRect().width;
+      const pageLayoutBox = this.element.getPageLayoutBox();
+      const whitespace = (viewportWidth - newWidth) / 2;
+      if (isRtl) {
+        newMargin = pageLayoutBox.right + whitespace - viewportWidth;
```

Comment:
newMargin could be a fraction of a pixel due to usage of division.  Please round to nearest int.

Reply:
Done.